specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+    ///   standard. If it does not, MultiToken contract's `mt_resolve_batch_transfer` MUST deal
+    ///   with the resulting failed cross-contract call and roll back the transfer.
+    /// * Contract MUST implement the behavior described in `mt_resolve_batch_transfer`
+    /// * `approval_id` is for use with Approval Management extension, see
+    ///   that document for full explanation.
+    /// * If using Approval Management, contract MUST nullify approved accounts on
+    ///   successful transfer.
+    ///
+    /// Arguments:
+    /// * `receiver_id`: the valid NEAR account receiving the token.
+    /// * `token_ids`: the tokens to transfer
+    /// * `amounts`: the amount of tokens to transfer for corresponding token_id
+    /// * `approval_ids`: expected approval IDs. A number smaller than
+    ///    2^53, and therefore representable as JSON. See Approval Management
+    ///    standard for full explanation. Must have same length as token_ids
+    /// * `memo` (optional): for use cases that may benefit from indexing or
+    ///    providing information for a transfer.
+    /// * `msg`: specifies information needed by the receiving contract in
+    ///    order to properly handle the transfer. Can indicate both a function to
+    ///    call and the parameters to pass to that function.
+
+    fn mt_batch_transfer_call(
+        &mut self,
+        receiver_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+        memo: Option<String>,
+        msg: String,
+    ) -> PromiseOrValue<Vec<U128>>;
+
+    /// Get the balance of an an account given token_id. For fungible token returns back amount, for
+    /// non fungible token it returns back constant 1.
+    fn mt_balance_of(&self, owner_id: AccountId, token_id: TokenId) -> U128;
+
+    /// Get the balances of an an account given token_ids. For fungible token returns back amount, for
+    /// non fungible token it returns back constant 1. returns vector of balances corresponding to token_ids
+    /// in a 1-1 mapping
+    fn mt_balance_of_batch(&self, owner_id: AccountId, token_ids: Vec<TokenId>) -> Vec<U128>;
+
+    /// Returns the total supply of the token in a decimal string representation given token_id.
+    fn mt_total_supply(&self, token_id: TokenId) -> U128;
+
+    // Returns the total supplies of the tokens given by token_ids in a decimal string representation.
+    fn mt_total_supply_batch(&self, token_ids: Vec<TokenId>) -> Vec<U128>;
+}
+```
+### Receiver Trait
+#### Notes
+- TokenId is of type String
+```
+pub trait MultiTokenReceiver {
+    /// Take some action after receiving a MultiToken-tokens token
+    ///
+    /// Requirements:
+    /// * Contract MUST restrict calls to this function to a set of whitelisted MultiToken
+    ///   contracts
+    ///
+    /// Arguments:
+    /// * `sender_id`: the sender of `mt_transfer_call`
+    /// * `previous_owner_id`: the account that owned the tokens prior to it being
```

Comment:
I see `previous_owner_id` listed here, but it doesn't appear in the function signature below?

Reply:
yeah its a relic of approval management, that was voted out. good catch.